sky/provision/aws/instance.py

```diff
@@ -791,3 +793,19 @@ def get_cluster_info(region: str,
         instances=instances,
         head_instance_id=head_instance_id,
     )
+
+
+def query_ports(
+    cluster_name_on_cloud: str,
+    ip: str,
+    ports: List[str],
+    provider_config: Optional[Dict[str, Any]] = None,
+) -> Dict[int, List[common.Endpoint]]:
+    """See sky/provision/__init__.py"""
+    del cluster_name_on_cloud, provider_config  # Unused.
+    ports = list(resources_utils.port_ranges_to_set(ports))
+    result: Dict[int, List[common.Endpoint]] = {}
+    for port in ports:
+        result[port] = [common.SocketEndpoint(host=ip, port=port)]
+
+    return result
```

Comment:
We seem to have the same implementation of `query_ports` for AWS, GCP and Azure. Should we create a common method that generates the result Dict and each cloud class can invoke it?